src/python/gudhi/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+nanobind_add_module_with_targets(_pers_cub_low_dim_ext _pers_cub_low_dim.cc TBB::tbb)
+nanobind_add_module_with_targets(_cubical_complex_ext _cubical_complex.cc TBB::tbb)
+nanobind_add_module_with_targets(_ripser_ext _ripser.cc)
+nanobind_add_module_with_targets(_rips_complex_ext _rips_complex.cc TBB::tbb)
+nanobind_add_module_with_targets(_edge_collapse_ext _edge_collapse.cc TBB::tbb)
+nanobind_add_module_with_targets(_simplex_tree_ext _simplex_tree.cc TBB::tbb)
+nanobind_add_module_with_targets(_reader_utils_ext _reader_utils.cc)
+nanobind_add_module_with_targets(_witness_complex_ext _witness_complex.cc TBB::tbb)
+nanobind_add_module_with_targets(_strong_witness_complex_ext _strong_witness_complex.cc TBB::tbb)
+nanobind_add_module_with_targets(_nerve_gic_ext _nerve_gic.cc TBB::tbb)
+nanobind_add_module_with_targets(_hera_ext _hera.cc)
+nanobind_add_module_with_targets(_bottleneck_ext _bottleneck.cc CGAL::CGAL TBB::tbb)
+if(TARGET Eigen3::Eigen)
+    add_library(_delaunay_complex_ext_2d SHARED _delaunay_complex_2d.cc)
+    add_library(_delaunay_complex_ext_3d SHARED _delaunay_complex_3d.cc)
+    add_library(_delaunay_complex_ext_dynd SHARED _delaunay_complex_dyn.cc)
+    # works only because NANOBIND_exec was already build by a call of `nanobind_add_module`
+    target_link_libraries(_delaunay_complex_ext_2d CGAL::CGAL Eigen3::Eigen $<TARGET_NAME_IF_EXISTS:TBB::tbb> ${NANOBIND_exec})
+    target_link_libraries(_delaunay_complex_ext_3d CGAL::CGAL Eigen3::Eigen $<TARGET_NAME_IF_EXISTS:TBB::tbb> ${NANOBIND_exec})
+    target_link_libraries(_delaunay_complex_ext_dynd CGAL::CGAL Eigen3::Eigen $<TARGET_NAME_IF_EXISTS:TBB::tbb> ${NANOBIND_exec})
+endif()
+nanobind_add_module_with_targets(_delaunay_complex_ext _delaunay_complex.cc Eigen3::Eigen CGAL::CGAL TBB::tbb _delaunay_complex_ext_2d _delaunay_complex_ext_3d _delaunay_complex_ext_dynd)
```

Comment:
IIUC, this builds 4 DLLs, where the last one depends on the first 3. Why not make it a single DLL?
```cmake
nanobind_add_module(_delaunay_complex_ext _delaunay_complex.cc _delaunay_complex_2d.cc _delaunay_complex_3d.cc _delaunay_complex_dyn.cc)
target_link_libraries(...)
```
?

Reply:
This modification is here to address [your comment](https://github.com/GUDHI/gudhi-devel/pull/1196#discussion_r2105865048) in order to reduce the risk of too long compilation with too much resources.

Reply:
I know, but my suggestion was (as in traditional C):
cc: file1.c -> file1.o
cc: file2.c -> file2.o
ld: *.o -> module.so
whereas the PR does
cc: file1.c -> file1.o
ld: file1.o -> file1.so
cc: file2.c -> file2.o
ld: file2.o -> file2.so with a runtime dependency on file1.so
In terms of compilation ("cc" lines), it is the same. In terms of linking ("ld"), it isn't. Unless we use LTO/LTCG (it doesn't look like it with the current CMakeLists.txt), the memory and speed issues are only during the compilation phase.

Reply:
I guess, I misunderstood something then. I thought you wanted also the `.so` to be smaller (i.e., to be split into smaller `.so`s as the sum won't indeed not be smaller). To avoid the memory use for the assembling.
But if having still a big `.so` is fine, 
```
nanobind_add_module(_delaunay_complex_ext _delaunay_complex.cc _delaunay_complex_2d.cc _delaunay_complex_3d.cc _delaunay_complex_dyn.cc)
target_link_libraries(_delaunay_complex_ext PRIVATE Eigen3::Eigen CGAL::CGAL TBB::tbb)
```
does work.

Reply:
I don't think there is much point splitting a .so into several .so that all get loaded together anyway.
The memory issue we had several times was during the compilation phase (.cc -> .o). Linking all the object files into a single DLL is much less likely to be a problem (at least if we don't use LTO).

Reply:
I prefer the one with the modification on `nanobind_add_module_with_targets`, and yes it will fix the memory issue at compilation time.
You can remove the commented lines (former version), and maybe add some comment to explain that these files are split because it was consuming too much memory at compilation.